nativelib/src/main/resources/scala-native/gc/shared/ThreadUtil.c

```diff
@@ -2,6 +2,20 @@
 #include <stdio.h>
 #include <limits.h>
 
+#if !(defined SCALANATIVE_GC_COMMIX ||                                         \
+      defined SCALANATIVE_MULTITHREADING_ENABLED)
+
+// If neither multithreading nor Commix GC is enabled, there's no concurrency
+// so we don't need to involve actual mutexes.
+
+INLINE
+bool mutex_init(mutex_t *ref) { return true; }
+
+INLINE bool mutex_lock(mutex_t *ref) { return true; }
+
+INLINE bool mutex_unlock(mutex_t *ref) { return true; }
+#else
```

Comment:
**question:** I only needed these three so I implemented only these. Perhaps we should do this on a per-function basis?